hyperopt/tests/test_spark.py

```diff
@@ -273,91 +273,86 @@ def test_parallelism_arg(self):
-        # Given invalidly small parallelism
-        with patch_logger("hyperopt-spark") as output:
-            parallelism = SparkTrials._decide_parallelism(max_num_concurrent_tasks, -1)
-            self.assertEqual(
-                parallelism,
-                max_num_concurrent_tasks,
-                "Failed to default parallelism ({p}) to max_num_concurrent_tasks"
-                " ({e})".format(p=parallelism, e=max_num_concurrent_tasks),
-            )
-            log_output = output.getvalue().strip()
-            self.assertIn(
-                "invalid value (-1)",
-                log_output,
-                """Invalid parallelism value -1 missing from log: {log_output}""".format(
-                    log_output=log_output
-                ),
-            )
-            self.assertIn(
-                "max_num_concurrent_tasks ({c})".format(c=max_num_concurrent_tasks),
-                log_output,
-                """max_num_concurrent_tasks value missing from log: {log_output}""".format(
-                    log_output=log_output
-                ),
+        for spark_default_parallelism, max_num_concurrent_tasks in [(2, 4), (2, 0)]:
+            default_parallelism = max(
+                spark_default_parallelism, max_num_concurrent_tasks
             )
 
-        # Given invalidly large parallelism
-        with patch_logger("hyperopt-spark") as output:
-            parallelism = SparkTrials._decide_parallelism(
-                max_num_concurrent_tasks, max_num_concurrent_tasks + 1
-            )
-            self.assertEqual(
-                parallelism,
-                max_num_concurrent_tasks,
-                "Failed to limit parallelism ({p}) to max_num_concurrent_tasks"
-                " ({e})".format(p=parallelism, e=max_num_concurrent_tasks),
-            )
-            log_output = output.getvalue().strip()
-            self.assertIn(
-                "parallelism ({p}) is greater".format(p=max_num_concurrent_tasks + 1),
-                log_output,
-                """User-specified parallelism ({p}) missing from log: {log_output}""".format(
-                    p=max_num_concurrent_tasks + 1, log_output=log_output
-                ),
-            )
-            self.assertIn(
-                "max_num_concurrent_tasks ({c})".format(c=max_num_concurrent_tasks),
-                log_output,
-                """max_num_concurrent_tasks value missing from log: {log_output}""".format(
-                    log_output=log_output
-                ),
-            )
-
-        # Given valid parallelism
-        parallelism = SparkTrials._decide_parallelism(max_num_concurrent_tasks, None)
-        self.assertEqual(
-            parallelism,
-            max_num_concurrent_tasks,
-            "The default parallelism ({p}) did not equal max_num_concurrent_tasks"
-            " ({e})".format(p=parallelism, e=max_num_concurrent_tasks),
-        )
+            # Test requested_parallelism is None or negative values.
```

Comment:
maybe a different test (same file but a different function) would help to localise where the tests are failing. If everything is in the same is not that easy. If we had pytest this would be a bit nicer using pytest.mark.parametrize....🤷‍♂️